spacy/scorer.py

```diff
@@ -473,17 +473,17 @@ def score_cats(
             threshold = 0.5 if multi_label else 0.0
         if not multi_label:
             threshold = 0.0
+        labels = set(labels)
         f_per_type = {label: PRFScore() for label in labels}
         auc_per_type = {label: ROCAUCScore() for label in labels}
-        labels = set(labels)
-        if labels:
```

Comment:
I think this was intended to be `if not labels` so it could infer the labels from the examples. I think the idea was that your reference docs might have some label that your component has completely missed and you want to include that in the evaluation, but that idea doesn't scale beyond one textcat component.